### PR TITLE
Removed USD component from SDFormat and move to gz-usd

### PR DESCRIPTION
**Pull request by ahcorde**
Signed-off-by: ahcorde <ahcorde@gmail.com>

# 🎉 New feature

Removed USD component from SDFormat and move to gz-usd

There are some reason to move this component: 

 - It's using the FUEL directory structure to get some models https://github.com/gazebosim/sdformat/issues/1051
 - External contributors asking for permission in SDFormat (easy to given then permissions in another repository)
 - Other converters are in a different repo https://github.com/gazebosim/gz-mujoco
 - This module is not yet released
 - Less time on CI

New location https://github.com/gazebosim/gz-usd

## Checklist
- [ ] Signed all commits for DCO
- [ ] Added tests
- [ ] Added example and/or tutorial
- [ ] Updated documentation (as needed)
- [ ] Updated migration guide (as needed)
- [ ] Consider updating Python bindings (if the library has them)
- [ ] `codecheck` passed (See [contributing](https://gazebosim.org/docs/all/contributing#contributing-code))
- [ ] All tests passed (See [test coverage](https://gazebosim.org/docs/all/contributing#test-coverage))
- [ ] While waiting for a review on your PR, please help review [another open pull request](https://github.com/pulls?q=is%3Aopen+is%3Apr+user%3Agazebosim+archived%3Afalse+) to support the maintainers

**Note to maintainers**: Remember to use **Squash-Merge** and edit the commit message to match the pull request summary while retaining `Signed-off-by` messages.

### CODE REVIEW

**Review by chapulina — CHANGES_REQUESTED** [review]
See this for the warning on Jenkins:

* https://github.com/gazebo-tooling/release-tools/pull/773

We should also update `examples/usdConverter` to point to the new location. I think it's worth it keeping the example here so the old links to them still work.

There's also the `test/usd` directory, and these:

* https://github.com/gazebosim/sdformat/blob/5a7544acd360bf37c09987cc3b4bc42133cb4083/.github/workflows/ci.yml#L20-L23
* https://github.com/gazebosim/sdformat/blob/5a7544acd360bf37c09987cc3b4bc42133cb4083/CMakeLists.txt#L113

Finally, let's add notes to the migration guide explaining what a user needs to change to use the new location.

File: Migration.md [hunk 1]
```diff
@@ -113,6 +113,9 @@ one root level model, actor or light.
    respectively. Users are encouraged to use the constants `kSdfStringSource` 
    and `kUrdfStringSource` instead of hard-coding the string literals.
 
+1. USD component now is living in https://github.com/gazebosim/gz-usd as an
+   independent package.
```
Comment (chapulina) [inline Migration.md:117]: Can we create a new section for this above, called `## libsdformat 12.5.0 to 12.6.0`?

File: examples/usdConverter/README.md [hunk 2]
```diff
@@ -6,33 +6,34 @@ This example shows how a world in a SDF file can be converted to [USD](https://g
 
 You will need all of the dependencies for sdformat, along with the following additional dependencies:
```
Comment (chapulina) [inline examples/usdConverter/README.md:7]: We should mention that SDFormat itself is also needed, now that cloning it isn't part of the tutorial anymore

**Comment by chapulina** [comment]
> the test/usd directory

That's still there. Remove them here and make sure the tests are also present in the new repo

File: examples/usdConverter/README.md (hunk 2 above)
Comment (ahcorde) [reply examples/usdConverter/README.md:7]: https://github.com/gazebosim/sdformat/pull/1094/commits/f3daf6e9c744d7d1fde87ccf13ca76695f79e008

File: Migration.md (hunk 1 above)
Comment (ahcorde) [reply Migration.md]: https://github.com/gazebosim/sdformat/pull/1094/commits/f3daf6e9c744d7d1fde87ccf13ca76695f79e008

**Review by chapulina — APPROVED** [review]

**Review by scpeters — APPROVED** [review]

**Pull request merged**